rskj-core/src/test/java/co/rsk/peg/lockingcap/LockingCapSupportImplTest.java

```diff
@@ -0,0 +1,302 @@
+
+        Coin maxLockingCap = previousLockingCap.multiply(constants.getIncrementsMultiplier());
+        Coin newLockingCap = maxLockingCap.add(Coin.SATOSHI);
+        Transaction tx = TransactionUtils.getTransactionFromCaller(signatureCache, LockingCapCaller.AUTHORIZED.getRskAddress());
+
+        // Act
+        boolean actualResult = lockingCapSupport.increaseLockingCap(tx, newLockingCap);
+
+        // Assert
+        assertFalse(actualResult);
+        assertEquals(Optional.of(previousLockingCap), lockingCapSupport.getLockingCap());
+    }
+
+    @Test
+    void increaseLockingCap_whenNewLockingCapIsEqualToMaxLockingCap_shouldReturnTrue() {
+        // Arrange
+        Coin previousLockingCap = constants.getInitialValue().add(Coin.SATOSHI);
+        lockingCapStorageProvider.setLockingCap(previousLockingCap);
+        lockingCapSupport.save();
+        // Recreate LockingCapSupport to load the previous value from storage
+        lockingCapStorageProvider = new LockingCapStorageProviderImpl(bridgeStorageAccessor);
+        lockingCapSupport = new LockingCapSupportImpl(lockingCapStorageProvider, activations, constants, signatureCache);
+
+        // The new locking cap is the maximum value that can be set
+        Coin newLockingCap = previousLockingCap.multiply(constants.getIncrementsMultiplier());
+        Transaction tx = TransactionUtils.getTransactionFromCaller(signatureCache, LockingCapCaller.AUTHORIZED.getRskAddress());
+
+        // Act
+        boolean actualResult = lockingCapSupport.increaseLockingCap(tx, newLockingCap);
+
+        // Assert
+        assertTrue(actualResult);
+        assertEquals(Optional.of(newLockingCap), lockingCapSupport.getLockingCap());
+    }
+
+    @Test
+    void increaseLockingCap_whenNewLockingCapIsEqualToLockingCapValueAlreadyExists_shouldReturnTrue() {
+        // Arrange
+        Coin newLockingCap = constants.getInitialValue();
+        Transaction tx = TransactionUtils.getTransactionFromCaller(signatureCache, LockingCapCaller.AUTHORIZED.getRskAddress());
+
+        // Act
+        boolean actualResult = lockingCapSupport.increaseLockingCap(tx, newLockingCap);
+
+        // Assert
+        assertTrue(actualResult);
+        assertEquals(Optional.of(constants.getInitialValue()), lockingCapSupport.getLockingCap());
+    }
+
+    @Test
+    void increaseLockingCap_whenNewLockingCapIsZero_shouldReturnFalse() {
+        // Arrange
+        Coin newLockingCap = Coin.ZERO;
+        Transaction tx = TransactionUtils.getTransactionFromCaller(signatureCache, LockingCapCaller.AUTHORIZED.getRskAddress());
+
+        // Act
+        boolean actualResult = lockingCapSupport.increaseLockingCap(tx, newLockingCap);
+
+        // Assert
+        assertFalse(actualResult);
+        assertEquals(Optional.of(constants.getInitialValue()), lockingCapSupport.getLockingCap());
+    }
+
+    @Test
+    void increaseLockingCap_whenNewLockingCapIsNegative_shouldReturnFalse() {
```

Comment:
same here. No restrictions about it being negative.

Reply:
Same here. It tests a different input and evaluates the behavior.

Reply:
same here. Not related to the method.

Reply:
Guys, I may not be following you both correctly, but I see two checks in the [implementation](https://github.com/rsksmart/rskj/blob/24b01df3639ec634ea55711604b9f1a1d72199a6/rskj-core/src/main/java/co/rsk/peg/lockingcap/LockingCapSupportImpl.java#L48), one for negative and one for zero value. So, why not adding tests for these cases?

Reply:
@julia-zack  is correct regarding this as well. In this case, I consider it does not hurt to test using these special values. Wdyt?

Reply:
They don't hurt, they support our tests to know the behavior of the `increaseLockingCap` method with different inputs. In the same line, we'll migrate the code from `Bridge.java` to `LockingCapSupportImpl.java` so these tests are necessary.